framework/validators/EmailValidator.php

```diff
@@ -96,7 +96,11 @@ protected function validateValue($value)
             } else {
                 $valid = preg_match($this->pattern, $value) || $this->allowName && preg_match($this->fullPattern, $value);
                 if ($valid && $this->checkDNS) {
-                    $valid = checkdnsrr($matches['domain'] . '.', 'MX') || checkdnsrr($matches['domain'] . '.', 'A');
+                    // Fix for https://github.com/yiisoft/yii2/issues/17083
+                    $valid = (
+                        (checkdnsrr($matches['domain'] . '.', 'MX') && count(dns_get_record($matches['domain'] . '.', DNS_MX)) > 0) ||
+                        (checkdnsrr($matches['domain'] . '.', 'A') && count(dns_get_record($matches['domain'] . '.', DNS_A)) > 0)
```

Comment:
@mikk150 Are you sure that this will work in all cases?
I mean `dns_get_record()` may return `false`, and since PHP 7.2 `count(false)` will throw warning (and before PHP 7.2 it will return 1, and I don't sure that it's better :) ).

Reply:
https://github.com/yiisoft/yii2/commit/7a82f7d88d46c539fe59ffc1d47e8541941ebdc8